src/components/Code/Code.js

```diff
@@ -0,0 +1,107 @@
+import React, { Component } from 'react';
+import styled from 'styled-components';
+import { isFunction } from 'lodash';
+
+const CopyNotice = styled.div`
+  position: absolute;
+  top: 5px;
+  right: 5px;
+  transition: opacity 300ms ease;
+  opacity: 0;
+  font-size: ${props => props.theme.fontSizes.tiny};
+  color: ${props => props.theme.colors.athensGray};
+`;
+
+const Root = styled.div`
+  position: relative;
+  padding: 10px 15px;
+  background-color: ${props => props.theme.colors.gunpowder};
+  box-sizing: border-box;
+  border-radius: ${props => props.theme.borderRadius.soft};
+  cursor: pointer;
+
+  &:hover ${CopyNotice} {
+    opacity: 1;
+  }
+`;
+
+const Content = styled.div`
+  flex-grow: 1;
+  overflow: auto;
+  color: ${props => props.theme.colors.athensGray};
+`;
+
+const Pre = styled.pre`
+  margin: 10px 0;
+  font-family: ${props => props.theme.fontFamilies.monospace};
+  font-size: ${props => props.theme.fontSizes.small};
+`;
+
+class Code extends Component {
```

Comment:
Component description (a comment above this line) that shows up in the _Examples_ page would also be nice.